feat: cyclopedia deaths

LeoTKBR opened the pull request:
Complement Commit: https://github.com/opentibiabr/canary/commit/e2952832789ce814ce98db6d1d5548d1f977b779

This is a continuation of the PR showing all the participants of the player's death, before it was made only for the website now it has been modified to appear in the client.

![image](https://github.com/user-attachments/assets/62c01042-d8d9-4cca-8dd6-9c5790da203b)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Death history now appends a full list of participating players and creatures (with counts like "2x Name") to cause descriptions.
  * Cause descriptions show both the killer and the top damage source, and explicitly mark true solo kills.

* **Bug Fixes / Tweaks**
  * Standardized level label to "Died at level" and improved spacing/formatting for clearer readability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

LeoTKBR committed:
Update login.lua

LeoTKBR committed:
Create autosync.yml

Auto Sync Fork

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Update autosync.yml

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'main' of https://github.com/LeoTKBR/canary

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
#Remove autosync

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
Merge branch 'opentibiabr:main' into main

LeoTKBR committed:
feat: cyclopedia deaths all participants

github-actions[bot] committed:
Code format - (Clang-format)

mdbeng commented:
Looks good to me, just one thing, it's better and more to what it actually is if the "participants" only shows 1 of them, is not necessary to say exactly all participants.

LeoTKBR commented:
> Looks good to me, just one thing, it's better and more to what it actually is if the "participants" only shows 1 of them, is not necessary to say exactly all participants.

This is a personal taste, each person will want it in a different way. I presented a simple way and the purpose is to show everything. If you want to modify it, feel free.

dudantas committed:
Merge branch 'main' into feat-cyclopedia-deaths

dudantas committed:
Merge branch 'main' into feat-cyclopedia-deaths

dudantas committed:
Update src/creatures/players/components/player_cyclopedia.cpp

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

dudantas committed:
Fix SQL query to count player deaths correctly

dudantas committed:
Improve participant entry parsing in player_cyclopedia

Refactor participant entry parsing to improve clarity and handle malformed entries.

dudantas committed:
Add function to trim carriage returns from names

Added a lambda function to trim carriage returns from names in player cyclopedia.

github-actions[bot] committed:
Code format - (Clang-format)

dudantas committed:
Update src/creatures/players/components/player_cyclopedia.cpp

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

dudantas committed:
Add logging for unrecognized participant types

github-actions[bot] committed:
Code format - (Clang-format)

dudantas committed:
Replace skipped.find() with skipped.contains()

dudantas committed:
Merge branch 'main' into feat-cyclopedia-deaths

dudantas committed:
Merge branch 'main' into feat-cyclopedia-deaths

dudantas committed:
Merge branch 'main' into feat-cyclopedia-deaths
